alacritty_config_derive/src/serde_replace.rs

```diff
@@ -88,19 +93,42 @@ fn match_arms<T>(fields: &Punctuated<Field, T>) -> TokenStream2 {
         let flatten = field
             .attrs
             .iter()
+            .filter(|attr| (*attr).path().is_ident("config"))
             .filter_map(|attr| attr.parse_args::<Attr>().ok())
             .any(|parsed| parsed.ident.as_str() == "flatten");
 
         if flatten && flattened_arm.is_some() {
-            return Error::new(ident.span(), MULTIPLE_FLATTEN_ERROR).to_compile_error();
```

Comment:
I'd assume this is what caused the errors previously, since the compile errors were converted too early?

Reply:
Yes, inside the match statement, where `replace_arms` where used, the expanded was something like this:
```rust
match field {
     "cursor" | "backround" => {},
      <WEIRD_FLATTEN_ERROR>
}
```

so the compiler was complaining that the match was incorrect